### ISSUE TITLE
Strange animation issue with route transition with #each

### ISSUE THREAD

**Issue by chrism** [ember-animation/liquid-fire#533]
I've made an animated gif which I hope demonstrates the issue as it is quite hard to explain.

![example_animation](https://cloud.githubusercontent.com/assets/35488/21185156/c979d546-c20f-11e6-9a00-f562aa3e638b.gif)

The route transitions (with a liquid-fire fade) but at the end, after the transition has completed, the height of the liquid-outlet jumps. It looks to me like the cause is that the data from the model actually only updates _after_ the transition.

I cannot work out why it is doing this though, because the data loads from a promise in the route, which looks like this:

```js
  model(params) {
    return this.store.findRecord('album', params.id);
  }
```

So surely it should block the transition until the album data has fully loaded?

I have made sure there are no async relationships that might be lazily loading. In fact I am munging the data from a 3rd party API into a tracklisting object on the model directly (with id and title) so that everything definitely loads in that one model request, and nothing should be updating in the background.

This is looped though via a simple `#each` in the template, ie

```hbs
<ol class="album-tracks__list">
  {{#each model.tracklisting as |track|}}
    <li class="album-tracks__list-item">{{link-to track.title "add-track.track" track.id}}</li>
  {{/each}}
</ol>
```

Is this a bug in the way that liquid-fire renders to an outlet, or have I missed something?

I'm really at a loss how to fix this.

Thanks.

**Comment by ef4** [comment]
My guess is that the layout changes when some of those images load. That could cause behavior like this. 

If the image dimensions are known in advance, you can set them directly via CSS and avoid the jump. Alternatively, there is a new-ish feature called `liquid-sync` that may help. Where you render an image, you would wrap it in liquid sync like:

```hbs
{{#liquid-sync as |readyToAnimate|}}
  <img src={{something}} onload=readyToAnimate onerror=readyToAnimate />
{{/liquid-sync}}
```

This declares to liquid fire that this whole area shouldn't be measured and animated until the image has loaded.

**Comment by chrism** [comment]
Oh wow, liquid-sync looks great!

Thanks very much for the explanation and I'm sorry but maybe the animation was a little bit misleading in that case.

Those images that you see are already loaded on a parent route, the only content that changes is the big image and the list of items (which is simple HTML/CSS no images)

- Ultimate Painting
- Can't You See?
- Central Park Blues
- Riverside
- Ten Street
- Rolling in the Deep End
- Three Piers
- Jane
- She's a Bomb
- Winter in Your Heart

I used the technique you mentioned already for the big image (I realised that this could have been the issue previously so set the width & height explicitly in the CSS).

Put simply, I think in this particular case it is not the images causing a problem therefore—I can make another animation with no images if that could be useful to you?

Thanks again.

**Comment by chrism** [comment]
Here is an animation to illustrate without any images.

![screenflow](https://cloud.githubusercontent.com/assets/35488/21188292/2f547d66-c21b-11e6-9b28-350bd08313a2.gif)

**Comment by chrism** [comment]
OK, so I have managed to work out what the issue is and it relates to the record being reloaded.

Because a partially loaded record for the album already exists it transitions immediately into the route, then loads the additional data from the additional request.

Therefore I was able to fix this issue by ensuring that the record always reloads, by using the `shouldReloadRecord` method in my adapter.

http://emberjs.com/api/data/classes/DS.JSONAPIAdapter.html#method_shouldReloadRecord

I'm closing this issue as it is not a liquid-fire bug, but hopefully might be useful to someone else experiencing this issue.

Thanks again for your advice.
